src/ResourceManager/Sql/Commands.Sql/TransparentDataEncryption/Cmdlet/SetAzureRmSqlManagedInstanceTransparentDataEncryptionProtector.cs

```diff
@@ -0,0 +1,143 @@
+
+            AzureRmSqlManagedInstanceTransparentDataEncryptionProtectorModel model = new AzureRmSqlManagedInstanceTransparentDataEncryptionProtectorModel(
+                resourceGroupName: this.ResourceGroupName,
+                managedInstanceName: this.ManagedInstanceName);
+
+            resultList.Add(ModelAdapter.GetAzureRmSqlManagedInstanceTransparentDataEncryptionProtector(model));
+
+            return resultList;
+        }
+
+        /// <summary>
+        /// Constructs the model to send to the update API
+        /// </summary>
+        /// <param name="model">The result of the get operation</param>
+        /// <returns>The model to send to the update</returns>
+        protected override IEnumerable<Model.AzureRmSqlManagedInstanceTransparentDataEncryptionProtectorModel> ApplyUserInputToModel(IEnumerable<Model.AzureRmSqlManagedInstanceTransparentDataEncryptionProtectorModel> model)
+        {
+            List<Model.AzureRmSqlManagedInstanceTransparentDataEncryptionProtectorModel> newEntity = new List<Model.AzureRmSqlManagedInstanceTransparentDataEncryptionProtectorModel>();
+
+            newEntity.Add(new Model.AzureRmSqlManagedInstanceTransparentDataEncryptionProtectorModel(
+                resourceGroupName: this.ResourceGroupName, 
+                managedInstanceName: this.ManagedInstanceName, 
+                type: this.Type, 
+                keyId: this.KeyId));
+
+            return newEntity;
+        }
+        
+        /// <summary>
+        /// Sends the TDE protector update request to the service
+        /// </summary>
+        /// <param name="entity">The update parameters</param>
+        /// <returns>The response object from the service</returns>
+        protected override IEnumerable<AzureRmSqlManagedInstanceTransparentDataEncryptionProtectorModel> PersistChanges(IEnumerable<AzureRmSqlManagedInstanceTransparentDataEncryptionProtectorModel> entity)
+        {
+            return new List<AzureRmSqlManagedInstanceTransparentDataEncryptionProtectorModel>() {
+                ModelAdapter.CreateOrUpdateManagedInstanceEncryptionProtector(entity.First())
+            };
+        }
+
+        /// <summary>
+        /// Entry point for the cmdlet
+        /// </summary>
+        public override void ExecuteCmdlet()
+        {
+            if (!this.MyInvocation.BoundParameters.ContainsKey("KeyId") && this.Type.Equals(EncryptionProtectorType.AzureKeyVault))
+            {
+                throw new PSArgumentException(
+                    string.Format(CultureInfo.InvariantCulture, Properties.Resources.KeyIdNotFoundForAzureKeyVaultEncryptionProtectorError),
+                    "KeyId");
+            }
+
+            if (ShouldProcess(this.KeyId))
+            {
+                if (Force || this.Type == EncryptionProtectorType.ServiceManaged || ShouldContinue(
```

Comment:
This if statement should be removed - you only need the ShouldProcess loop

Reply:
customer will be presented with the warning only when they set an AKV key as encryption protector, so if condition is required

Reply:
Please make this this.Type != EncryptionProtectorType.AzureKeyVault instead, in case there are additional types added later.  Also that's more clear about why the warning is being shown.